openvasd/openvasd.c

```diff
@@ -1591,6 +1589,56 @@ openvasd_get_health_started (openvasd_connector_t conn)
   return response;
 }
 
+openvasd_resp_t
+openvasd_get_performance (openvasd_connector_t conn,
+                          openvasd_get_performance_opts_t opts)
+{
+  openvasd_resp_t response = NULL;
+  gchar *err = NULL;
+  CURL *hnd = NULL;
+  struct curl_slist *customheader = NULL;
+  time_t now;
+  time (&now);
+
+  response = g_malloc0 (sizeof (struct openvasd_response));
+
+  if (!opts.titles || !strcmp (opts.titles, "") || opts.start < 0
+      || opts.start > now || opts.end < 0 || opts.end > now)
+    {
+      response->code = RESP_CODE_ERR;
+      response->body =
+        g_strdup ("{\"error\": \"Couldn't send get_performance command "
+                  "to scanner. Bad or missing parameters.\"}");
+      return response;
+    }
+  gchar *query =
+    g_strdup_printf ("/health/performance?start=%d&end=%d&titles=%s",
+                     opts.start, opts.end, opts.titles);
+  customheader = init_customheader (conn->apikey, FALSE);
+  hnd = handler (conn, GET, query, NULL, customheader, &err);
+  if (hnd == NULL)
+    {
+      curl_slist_free_all (customheader);
+      response->code = RESP_CODE_ERR;
+      response->body = err;
+      return response;
+    }
+
+  openvasd_send_request (hnd, NULL, response);
+  curl_slist_free_all (customheader);
+  if (response->code != RESP_CODE_ERR)
+    response->body = g_strdup (openvasd_vt_stream_str (conn));
+  else if (response->code == RESP_CODE_ERR)
```

Comment:
You could drop the `if` and just have the `else` because the condition is always true.

Reply:
Hmm, this is like this everywhere in this file. I'm happy to do them all as a PR.